Src/Couchbase.Linq.Tests/QueryGeneration/NestTests.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+using Couchbase.Core;
+using Couchbase.Linq.Extensions;
+using Couchbase.Linq.Tests.Documents;
+using Moq;
+using NUnit.Framework;
+
+namespace Couchbase.Linq.Tests.QueryGeneration
+{
+    [TestFixture]
+    public class NestTests : N1QLTestBase
+    {
+        [Test]
+        public void Test_Unnest_Simple()
+        {
+            var mockBucket = new Mock<IBucket>();
+            mockBucket.SetupGet(e => e.Name).Returns("default");
+
+            var query = from brewery in QueryFactory.Queryable<Brewery>(mockBucket.Object)
+                        from address in brewery.Address
+                        select new {name = brewery.Name, address};
+
+            const string expected = "SELECT brewery.name as name, address as address " +
+                "FROM default as brewery "+
+                "INNER UNNEST brewery.address as address";
+
+            var n1QlQuery = CreateN1QlQuery(mockBucket.Object, query.Expression);
+
+            Assert.AreEqual(expected, n1QlQuery);
+        }
+
+        [Test]
+        public void Test_Unnest_Sort()
+        {
+            var mockBucket = new Mock<IBucket>();
+            mockBucket.SetupGet(e => e.Name).Returns("default");
+
+            var query = from brewery in QueryFactory.Queryable<Brewery>(mockBucket.Object)
```

Comment:
I am not fond of the syntax, even though I know you don't have much choice given UNNEST has no LINQ equivalent. I am wondering if it's something we could add with Roslyn (new keywords: NEST and UNNEST)?

If we continued on this route, what would NEST look like?